--- a/Classes/Generator/SqlGenerator.php
+++ b/Classes/Generator/SqlGenerator.php
@@ -72,6 +72,7 @@
     {
         $indexes = [];
         $fields = [];
+        $foreignFields = [];
         $table = $tableDefinition->table;
         foreach ($tableDefinition->parentReferences as $parentReference) {
             $index = [];
@@ -89,6 +90,7 @@
             if (isset($parentTcaConfig['foreign_field'])) {
                 $foreignField = $parentTcaConfig['foreign_field'];
                 $index[] = $foreignField;
+                $foreignFields[] = $foreignField;
             }
             $indexes[] = $index;
         }
@@ -99,6 +101,12 @@
                 $sql[] = $sqlStatement;
             }
         }
+        foreach ($foreignFields as $foreignFieldName) {
+            $sqlStatement = 'CREATE TABLE `' . $table . '` (`' . $foreignFieldName . '` int(11) UNSIGNED DEFAULT \'0\' NOT NULL);';
+            if (!in_array($sqlStatement, $sql, true)) {
+                $sql[] = $sqlStatement;
+            }
+        }
         $indexes = array_map(fn(array $index): string => implode(', ', $index), $indexes);
         $indexes = array_unique($indexes);
         $indexes = array_values($indexes);
